url647b55da7c_Nikhilkumar13_emuSr_tgz-pJ8-MainJ8/src/backend/step_out.java

```diff
@@ -0,0 +1,220 @@
+package backend;
+
+import frontend.FrontEnd;
+import frontend.handlers;
+import java.awt.Color;
+import java.io.File;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.util.StringTokenizer;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.swing.text.BadLocationException;
+
+/**
+ *
+ * @author Tushar
+ */
+public class step_out implements Runnable
+{
+
+//    public static boolean barrelshiftpresent = false;
+    public static Long pos = new Long(0);
+    public static RandomAccessFile br = null;
+    public static Integer curr_line;
+    public static int common;
+    public Thread t;
+    public ScanFile ob;
+    public File fpath;
+    public static boolean call;
+    public static int initial_pos;
+//    public static int end_pos;
+    public static int next;
+//    private int ld;
+    public static int i = 0;
+    public static int j = 0;
+    private int abc = 0;
+    int exitCondition;
+    //private static boolean commentflag = false;
+
+    public step_out(File path, int ini, int fin, int f)
+      {
+
+        t = new Thread(this, "stepout");
+//        t1 = new Thread(this, "step");
+        fpath = path;
+        exitCondition = ScanFile.call_count;
+        if (f == 0 && ini != 0)
+        {
+
+            StepRun.ini_line = ini;
+            StepRun.fin_line = fin;
+            if (ini == fin)
+            {
+                StepRun.ini_line = 1;
+            } else
+            {
+                StepRun.ini_line = ini;
+            }
+            //System.out.println("BLAH " + StepRun.ini_line + "BLAH" + StepRun.fin_line + "BLAH" + f);
+        }
+
+        f = 1;
+      }
+//    static String[] shifts = {"ROR", "LSL", "LSR", "RRX", "ASR"};
+
+    
+    
+
+    @Override
+    public void run()
+      {
+        try
+        {
+            //System.out.println("step_out run()");
+            scanStart();
+        } catch (InterruptedException ex)
+        {
+            Logger.getLogger(step_out.class.getName()).log(Level.SEVERE, null, ex);
+        }
+      }
+
+    public static boolean branchToProcedure(String strLine) throws BadLocationException
+      {
+
+        if (new StringTokenizer(strLine, " ,\t").countTokens() == 2 && strLine.toUpperCase().contains("BL"))
+        {
+            //  frontend.FrontEnd.activepane.getHighlighter().addHighlight(initial_pos, end_pos, frontend.handlers.cyanPainter);
+            return true;
+        }
+        return false;
+      }
+
+   
+
+    
+    
+
+    public void scanStart() throws InterruptedException
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 15. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.